src/DynamoCore/Core/AuthenticationManager.cs

```diff
@@ -33,11 +34,31 @@ public LoginState LoginState
             get { return HasAuthProvider ? authProvider.LoginState : LoginState.LoggedOut; }
         }
 
-        internal bool IsLoggedIn()
+        /// <summary>
+        /// This Property will return the value by checking only once time to the real time LoginState in order to be used in the Initialization flow. Others features require direct calls to the LoginState due to are on demand.
+        /// </summary>
+        public LoginState? LoginStateSingle
         {
+            get
+            {
+                if (singleState == null)
+                {
+                    singleState = LoginState;
+                }
+                return singleState;
+            }
+        }
+
+        internal bool IsLoggedIn()
+        {            
             return HasAuthProvider && authProvider.LoginState == LoginState.LoggedIn ? true : false;
         }
 
+        internal bool IsLoggedInSingle()
```

Comment:
Can you put comments on what this property means?

Reply:
Done.